### PR TITLE
Refactoring human limb icon generation.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
## Description of changes
- Simplifies human icon gen to use bodypart keys instead of generating a bespoke key.
- Simplifies limb icon gen to minimize extra icon ops and properly use caching.
- Generalizes some overlay and marking handling.

## Why and what will this PR improve
Cleaner code. Probably slightly faster? Needed for #3191.

## Authorship
Myself.

## Changelog
Nothing player-facing.

### CODE REVIEW

File: code/game/objects/alien_props.dm [hunk 1]
```diff
@@ -8,13 +8,13 @@
 	icon_state = "unknown1"
 	maxcharge = 5000
 	origin_tech = "{'powerstorage':7}"
-	var/static/base_icon
+	var/base_icon_state
```
Comment (out-of-phaze) [inline code/game/objects/alien_props.dm:11]: I don't understand why this change was made, I think the original idea was that all alien cells in a round would have the same random icon. If the intent is to change that, that's fine I guess, I just don't know why it's in this PR.

Comment (out-of-phaze) [reply code/game/objects/alien_props.dm:11]: Oh, I see. Every instance of `base_icon` was audited.

**Review by out-of-phaze — DISMISSED** [review]

File: code/game/objects/alien_props.dm (hunk 1 above)
Comment (MistakeNot4892) [reply code/game/objects/alien_props.dm:11]: Yeah sorry, I wanted to make sure I got every instance so renamed for clarity.

**Review by out-of-phaze — DISMISSED** [review]

**Review by out-of-phaze — DISMISSED** [review]

**Review by out-of-phaze — APPROVED** [review]

**Review by deathride58 — APPROVED** [review]
